src/coreclr/jit/importer.cpp

```diff
@@ -9538,68 +9476,35 @@ GenTree* Compiler::impFixupCallStructReturn(GenTreeCall* call, CORINFO_CLASS_HAN
     else
     {
 
-#if FEATURE_MULTIREG_RET
+#if !FEATURE_MULTIREG_RET
+        return call;
+#else  // FEATURE_MULTIREG_RET
         const ReturnTypeDesc* retTypeDesc = call->GetReturnTypeDesc();
         const unsigned        retRegCount = retTypeDesc->GetReturnRegCount();
         assert(retRegCount != 0);
-        if (!compDoOldStructRetyping() && retRegCount == 1)
+        if (retRegCount == 1)
         {
             return call;
         }
-#else  // !FEATURE_MULTIREG_RET
-        if (!compDoOldStructRetyping())
-        {
-            return call;
-        }
-#endif // !FEATURE_MULTIREG_RET
-        assert(returnType != TYP_UNKNOWN);
 
-        // See if the struct size is smaller than the return
-        // type size...
-        if (howToReturnStruct == SPK_EnclosingType)
-        {
-            // If we know for sure this call will remain a call,
-            // retype and return value via a suitable temp.
-            if ((!call->CanTailCall()) && (!call->IsInlineCandidate()))
-            {
-                call->gtReturnType = returnType;
-                return impAssignSmallStructTypeToVar(call, retClsHnd);
-            }
-        }
-        else
-        {
-            // Return type is same size as struct, so we can
-            // simply retype the call.
-            call->gtReturnType = returnType;
-        }
+        assert(returnType == TYP_STRUCT);
+        assert((howToReturnStruct == SPK_ByValueAsHfa) || (howToReturnStruct == SPK_ByValue));
```

Comment:
This is is a possible change in the functionality of this method before we were handling SPK_EnclosingType, now we don't.

Reply:
SPK_EnclosingType becomes unreachable without `compDoOldStructRetyping`, I have added an assert that only `SPK_ByValueAsHfa` and `SPK_ByValue` can come here.

We used `SPK_EnclosingType ` only for structs that are smaller than TARGET_POINTER_SIZE and we were retyping them as int/long, now we keep them as structs, so now need for this part.

The part that is left handles the multi-reg case.